src/rules/index.ts

```diff
@@ -103,28 +112,39 @@ export class Rules {
     graph: DependencyGraph,
     resultListener: ResultListener<RuleResult>,
   ): Promise<RuleResult> {
-    const context = {
-      rule: async (targetId: string) => {
-        // Resolve aliases.
-        if (this.aliases) {
-          const alias = this.aliases.get(targetId);
-
-          if (alias !== undefined) {
-            targetId = alias;
+    const evaluator = new Evaluator({
+      // Build up context using custom functions and various defaults.
+      context: {
+        rule: async (targetId: string) => {
+          // Resolve aliases.
+          if (this.aliases) {
+            const alias = this.aliases.get(targetId);
+
+            if (alias !== undefined) {
+              targetId = alias;
+            }
           }
-        }
 
-        // First check if it will create a circular dependency.
-        // This throws an error if it will otherwise stores it.
-        graph.addDependency(id, targetId);
-        const ruleResult = await resultListener.wait(targetId);
-        return ruleResult.value;
+          // First check if it will create a circular dependency.
+          // This throws an error if it will otherwise stores it.
+          graph.addDependency(id, targetId);
+          const ruleResult = await resultListener.wait(targetId);
+          return ruleResult.value;
+        },
+        Math: MathContext,
+        String: StringContext,
+        Convert: ConvertContext,
```

Comment:
I think I would've preferred a few alternates instead of just toString and toNumber. Allow things like S() N() String()

Reply:
I think that isn't very clear at all what is going on, especially the single letter ones. Adding `String` and `Number` to the global context might be alright, since that's closer to JS, but I don't see the point on adding either single letter ones, or the JS-like ones under `Convert` (since it wouldn't make any sense to use them)

Reply:
Laziness, you love laziness. I can settle for just String and Number in the global context though